pygmt/datasets/earth_relief.py

```diff
@@ -54,8 +99,23 @@ def load_earth_relief(resolution="01d", registration=None):
             "gridline-registered grid is available."
         )
 
-    fname = which(f"@earth_relief_{resolution}{reg}", download="a")
-    grid = xr.open_dataarray(fname)
+    # different ways to load tiled and non-tiled earth relief data
+    if resolution in non_tiled_resolutions:
+        fname = which(f"@earth_relief_{resolution}{reg}", download="a")
+        with xr.open_dataarray(fname) as dataarray:
+            grid = dataarray.load()
+            _ = grid.gmt  # load GMTDataArray accessor information
+    elif resolution in tiled_resolutions:
+        # Titled grid can't be sliced.
+        # See https://github.com/GenericMappingTools/pygmt/issues/524
+        if region is None:
+            raise GMTInvalidInput(
+                "region is required for high resolution (<='05m') Earth relief grid"
+            )
+        grid = grdcut(f"@earth_relief_{resolution}{reg}", region=region)
+    else:
+        raise GMTInvalidInput(f"Invalid Earth relief resolution '{resolution}'")
```

Comment:
>     2. There is a styling error which I don't understand.

So it's a `flake8` error saying: `pygmt/datasets/earth_relief.py:100:17: F541 f-string is missing placeholders`. It doesn't error for me locally, could you run `make check` though? The tiled_resolutions list will turn into a single line, and that line 100 is this one. If I'm reading https://stackoverflow.com/questions/62565979/github-action-flake8-fails-f-string-is-missing-placeholders correctly, you'll need to change it to this:
```suggestion
        raise GMTInvalidInput(f'Invalid Earth relief resolution "{resolution}"')
```